m3/contract/filecoin.mock.sol

```diff
@@ -0,0 +1,671 @@
+        CommonTypes.Addr owner;
+        CommonTypes.Addr worker;
+        CommonTypes.Addr[] control_addresses;
+    }
+    struct GetOwnerReturn {
+        CommonTypes.Addr owner;
+    }
+    struct IsControllingAddressParam {
+        CommonTypes.Addr addr;
+    }
+    struct IsControllingAddressReturn {
+        bool is_controlling;
+    }
+    struct GetSectorSizeReturn {
+        CommonTypes.SectorSize sector_size;
+    }
+    struct GetAvailableBalanceReturn {
+        int256 available_balance;
+    }
+    struct GetVestingFundsReturn {
+        CommonTypes.VestingFunds[] vesting_funds;
+    }
+    struct ChangeWorkerAddressParams {
+        CommonTypes.Addr new_worker;
+        CommonTypes.Addr[] new_control_addresses;
+    }
+    struct ChangePeerIDParams {
+        bytes new_id;
+    }
+    struct ChangeMultiaddrsParams {
+        bytes new_multi_addrs;
+    }
+    struct SubmitWindowedPoStParams {
+        uint64 deadline;
+        CommonTypes.PoStPartition[] partitions;
+        CommonTypes.PoStProof[] proofs;
+        int64 chain_commit_epoch;
+        bytes chain_commit_rand;
+    }
+    struct ProveCommitAggregateParams {
+        uint8 sector_numbers;
+        bytes aggregate_proof;
+    }
+    struct ProveReplicaUpdatesParams {
+        CommonTypes.ReplicaUpdate[] updates;
+    }
+    struct ProveReplicaUpdatesParams2 {
+        CommonTypes.ReplicaUpdate2 updates;
+    }
+    struct DisputeWindowedPoStParams {
+        uint64 deadline;
+        uint64 post_index;
+    }
+    struct PreCommitSectorParams {
+        CommonTypes.RegisteredSealProof seal_proof;
+        uint64 sector_number;
+        CommonTypes.CID sealed_cid;
+        int64 seal_rand_epoch;
+        uint64[] deal_ids;
+        int64 expiration;
+        bool replace_capacity;
+        uint64 replace_sector_deadline;
+        uint64 replace_sector_partition;
+        uint64 replace_sector_number;
+    }
+
+    struct PreCommitSectorBatchParams {
+        PreCommitSectorParams[] sectors;
+    }
+
+    struct PreCommitSectorBatchParams2 {
+        CommonTypes.SectorPreCommitInfo[] sectors;
+    }
+
+    struct ProveCommitSectorParams {
+        uint64 sector_number;
+        bytes proof;
+    }
+
+    struct ConfirmSectorProofsParams {
+        uint64[] sectors;
+        CommonTypes.FilterEstimate reward_smoothed;
+        int256 reward_baseline_power;
+        CommonTypes.FilterEstimate quality_adj_power_smoothed;
+    }
+    struct CheckSectorProvenParams {
+        uint64 sector_number;
+    }
+    struct ExtendSectorExpirationParams {
+        CommonTypes.ExpirationExtension[] extensions;
+    }
+
+    struct ExtendSectorExpiration2Params {
+        CommonTypes.ExpirationExtension2[] extensions;
+    }
+
+    struct TerminateSectorsParams {
+        CommonTypes.TerminationDeclaration[] terminations;
+    }
+
+    struct TerminateSectorsReturn {
+        bool done;
+    }
+
+    struct DeclareFaultsParams {
+        CommonTypes.FaultDeclaration[] faults;
+    }
```

Comment:
Is there a reason you're including these "Later" methods in the library? I expected to only see the methods marked as "Export" in the [tracking sheet](https://docs.google.com/spreadsheets/d/1McCcCDHLagBFd5miTcTDBPuTIzmY6uKTxX6SS-xsKVY/edit#gid=0).

Reply:
No reason raul! I just took every method and mocked it, as it was not hard to do. But I will remove them, or keep them in another branch for now.

Reply:
I didn't realize "Export" ones were so few. I will keep the rest of them on another branch, so we can use it later when they are necessary.

Reply:
@raulk I have removed what it not required now.